### ISSUE TITLE
Regression Unable to import Quick Test if no user data is included.

### ISSUE THREAD

**Issue by eilers** [corona-warn-app/cwa-app-ios#3441]
There seems to be a regression:
Importing of Quick test results via QR-Code and via direct link stopped working. But only if there is no user data added. It still works if user name and birth is included (see attached QR-Codes). 
We will see the following error message: "Der QR-Code kann keinem Schnelltest zugeordnet werden. Bitte scannen Sie einen geeigneten QR-Code."
The generation of the QR-Code does not change since months, therefore I expect this to be a regression of the CWA.

I will check whether the Android version has the same problem as soon as possible. Unfortunately I do not have access to such a device now.

## Technical details

- Device name: iPhone 12
- iOS version: 14.7.1
- App version: 2.8.1

## Steps to reproduce the issue
Please scan the following code and you will see the message "Der QR-Code kann keinem Schnelltest zugeordnet werden. Bitte scannen Sie einen geeigneten QR-Code." No further information is provided.

The Code **(Note: Already used!! See new code below!):** ![qrcode](https://user-images.githubusercontent.com/115845/131373340-7c0223e0-7e7b-4094-991f-020ec7aef865.png)

The following code does work (it contains user data) **(Note: Already used!! See new code below!):**:
![qrcode](https://user-images.githubusercontent.com/115845/131373598-bdc413e0-4d1c-43f1-a079-5a78dc5c2cbb.png)

---
Internal Tracking ID: [EXPOSUREAPP-9258](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-9258)

**Comment by Ein-Tim** [comment]
This seems to be closely related to https://github.com/corona-warn-app/cwa-quicktest-onboarding/issues/55 where the same problem happened, however, only with test results with user data included.
Ping @vaubaehn

**Comment by vaubaehn** [comment]
@eilers 
Did you check whether there had been recent changes on server implementations (cwa-quicktest-backend, cwa-testresult-server)?

**Comment by vaubaehn** [comment]
I was able to import the RAT without personalized data into CWA 2.6.1 Android (first QR code). This test should now be "burnt" and the first QR code should succeed in scanning again.
It failed for the RAT with personalized data (second QR code) for the same device. So, I can confirm your results also with Android, but with CWA 2.6.1, so it shouldn't be any regression.
Both decoded JSONs look fine.
Given, that noone scanned the personalized QR code before me, seems to be a bug.

When you didn't change anything on your backend, then I'm also a bit clueless, what could be the reason.
You always double-hashed the test id for posting results to the server, didn't you?

**Comment by vaubaehn** [comment]
@eilers 
Ah, sorry, I got it mixed up!
Yes, the first code worked on Android, the second didn't (likely because you scanned it before) - so this bug seems to be iOS specific!

**Comment by eilers** [comment]
> Yes, the first code worked on Android, the second didn't (likely because you scanned it before) - so this bug seems to be iOS specific!

Thanks for testing Android. Someone has to look more closely what happens on iOS, then.  I've created a new QR-Code for further testing on iOS:
![qrcode](https://user-images.githubusercontent.com/115845/131454780-1c42f175-b834-45be-9435-27811428176b.png)

It shows the same error:
![IMG_FF6992CF4D70-1](https://user-images.githubusercontent.com/115845/131454835-b01ee247-6e67-46dd-ab84-fa9c5c11616e.jpeg)


> (Thanks for the negative test, btw... Although I'm vacc'ed)
You're welcome! :D

**Comment by vaubaehn** [comment]
@eilers 
Did you push the test result with the same test id in the morning? Because I just saw by chance, that the yesterday's imported test has now an updated `sc` timestamp of today!
I think, this is not intended behaviour, is it, @thomasaugsten ?

![Screenshot_2021-08-31-16-05-08](https://user-images.githubusercontent.com/68376758/131517812-f3fa87f4-edb6-4ad7-ad20-0965996014da.png)

**Comment by dsarkar** [comment]
- Using https://github.com/corona-warn-app/cwa-app-ios/issues/3441#issuecomment-908946556

- To summarise what I could reproduce:

![image](https://user-images.githubusercontent.com/1409741/131531975-8d1e4431-659a-4b14-b727-42c6bfc66ec9.png)

**Comment by vaubaehn** [comment]
@dsarkar 
Just to be sure: You started with iOS, and then Android, right? It looks like someone else already scanned QR, because you should not have gotten any error on Android...

**Comment by vaubaehn** [comment]
@eilers 
Does the error only occur, when you have POSTed a test result _before_ the user registered the test in CWA?
Or does the error also occur, when the user registers the test and you didn't submit any result yet?

**Comment by dsarkar** [comment]
@vaubaehn yes, I get an error, which is expected, so far, so good. I tested with the QR code and link given in https://github.com/corona-warn-app/cwa-app-ios/issues/3441#issuecomment-908946556 (no user data included)

1. Android - scan with CWA -> "... ungültig oder schon registriert ..."
1. Android - scan with third party app -> "... ungültig oder schon registriert ..."
1. Android - click on link -> "... ungültig oder schon registriert ..."
1. iOS - scan with CWA -> "... ungültig oder schon registriert ..."
1. iOS - scan with native camera -> "... kann keinen Schnelltest zugeordnet werden..."
1. iOS - click on link -> "... kann keinen Schnelltest zugeordnet werden..."

- I think, we can expect 1 - 4. 
- 5 & 6 are a bug (?)

**Comment by vaubaehn** [comment]
@dsarkar 
Ha... Nothing can be concluded from your experiment, unfortunately:
To be able to see that there is a bug on iOS, we need a fully working _unscanned_ test.
1. Try to import to iOS -> see error. But the test should still be functional/unused.
2. Try to import into Android -> there should be no error at all. Test is now used/burnt.
3. Any other import should not succeed anymore, as test is burnt now.

**Comment by vaubaehn** [comment]
@dsarkar 
What you could try: register an unpersonalized test with dm (NETSYNO). Does this work on iOS?
https://corona-schnelltest-zentren.dm.de/o/dm/login

**Comment by Ein-Tim** [comment]
@vaubaehn 

I want to try if I can reproduce this using the link you provided. Should I scan the QR-Code or click on the link?

**Comment by vaubaehn** [comment]
@Ein-Tim 

> I want to try if I can reproduce this using the link you provided. Should I scan the QR-Code or click on the link?

If there is no bug in iOS, you can do either/or. After you did one of them, the other shouldn't work anymore (because the test  should then be registered and be burnt for other use). But if there is actually a bug in iOS, then neither/nor will work and you will always get an error.

**Comment by Ein-Tim** [comment]
@vaubaehn 

Thanks, just tried it. Works when clicking on the link, scanning with CWA as well as scanning with the native camera.

So, conclusion, sorry, but IHMO, there must be a flaw in your implementation @eilers 😅

**Comment by eilers** [comment]
@vaubaehn
> Did you push the test result with the same test id in the morning? Because I just saw by chance, that the yesterday's imported test has now an updated `sc` timestamp of today!
> I think, this is not intended behaviour, is it, @thomasaugsten ?
You are absolutely right. This was a bug. I've fixed it, but unfortunately it does not fixed this issue.. 
But thanks for your hint!

**Comment by eilers** [comment]
@vaubaehn 
> 
> Thanks, just tried it. Works when clicking on the link, scanning with CWA as well as scanning with the native camera.
> 
> So, conclusion, sorry, but IHMO, there must be a flaw in your implementation @eilers 😅

Maybe it is a bug on my side.. But I still don't know _what_ is the problem.. 🙂  
Due to the fact that my test cases were always burned by Android (which doesn seem to have a problem to import it) I don't see that anyone could really check why they do not work on iOS!?

Ok, let's do a last try:
**PLEASE DO NOT USE THIS QR-Code on Android!! This is only for iOS!!! **
The following QR code could not be imported by my CWA on **iOS**.
![qrcode](https://user-images.githubusercontent.com/115845/132202340-c0197e76-8129-4bc5-a232-a43cd3d7b289.png)

The import link: https://s.coronawarn.app/?v=1#eyJ0aW1lc3RhbXAiOjE2MzA5MjI5NDYsInNhbHQiOiJBRTgxNDdCMUFEQTlENjExOUY0M0Y2OTA4RTlDN0NGQSIsImhhc2giOiI0OWE2ODRkMDA3ZGI5OTM1NTQyODA4YzU0ZDExNjA2MWZkNjU4YWFiYzQyYzcwMmE3ZDZkNDIwMzBlZWQ2NjAxIn0

Thanks a lot! 🙂

**Comment by eilers** [comment]
Ok, lets summarise the facts as we know:
1. The QR-Code is importable by CWA on Android and works (do not check it please, or you will burn the test case!)
2. The QR-Code is not accepted by CWA on iOS..

Anyone out there to look in to the reason for this?

Thanks in advance!

**Comment by eilers** [comment]
> @vaubaehn yes, I get an error, which is expected, so far, so good. I tested with the QR code and link given in [#3441 (comment)](https://github.com/corona-warn-app/cwa-app-ios/issues/3441#issuecomment-908946556) (no user data included)
> 
> OS	Method	Notification
> 1	Android	scan with CWA	"... ungültig oder schon registriert ..."
> 2	Android	scan with third party app	"... ungültig oder schon registriert ..."
> 3	Android	click on link	"... ungültig oder schon registriert ..."
> 4	iOS	scan with CWA	"... ungültig oder schon registriert ..."
> 5	iOS	scan with native camera	"... kann keinen Schnelltest zugeordnet werden..."
> 6	iOS	click on link	"... kann keinen Schnelltest zugeordnet werden..."
> * I think, we can expect 1 - 4.
> * 5 & 6 are a bug (?)
I see with a fresh qr-code:

OS	Method	Notification

**Comment by dsarkar** [comment]
@eilers Yes, we will have a look at this.

**Comment by Ein-Tim** [comment]
I assume this was closed by accident?

**Comment by eilers** [comment]
> I assume this was closed by accident?

Yes.. That was not expected.

**Comment by Ein-Tim** [comment]
Can you reopen it @eilers? If not maybe @dsarkar could reopen it 🙂

**Comment by eilers** [comment]
> Can you reopen it @eilers? If not maybe @dsarkar could reopen it 🙂

oops. sure..

**Comment by dsarkar** [comment]
@eilers Thanks for the feedback.

Best wishes, DS

---
Corona-Warn-App Open Source Team
